pollard_test.go

```diff
@@ -1339,3 +1340,148 @@ func compareNodeMap(mapA, mapB map[miniHash]*polNode) error {
 
 	return fmt.Errorf(str)
 }
+
+func TestCachedNodesAfterDelete(t *testing.T) {
+	// Define the test cases as a table.
+	tests := []struct {
+		name            string
+		numAdds         uint64
+		numDels         int
+		delIndices      []uint64
+		rememberIndices []uint64
+	}{
+		{
+			name:            "delete one node",
+			numAdds:         10,
+			numDels:         1,
+			delIndices:      []uint64{4},
+			rememberIndices: []uint64{1, 3, 4, 5, 6},
+		},
+		{
+			name:            "delete two nodes",
+			numAdds:         10,
+			numDels:         2,
+			delIndices:      []uint64{3, 7},
+			rememberIndices: []uint64{1, 3, 5, 7, 9},
+		},
+		{
+			name:            "delete three nodes",
+			numAdds:         10,
+			numDels:         3,
+			delIndices:      []uint64{0, 1, 7},
+			rememberIndices: []uint64{0, 1, 2, 3, 4, 7, 8, 9},
+		},
+		{
+			name:            "delete four nodes",
+			numAdds:         10,
+			numDels:         4,
+			delIndices:      []uint64{0, 1, 2, 3},
+			rememberIndices: []uint64{0, 1, 2, 3, 4, 5, 8, 9},
+		},
+		{
+			name:            "delete five nodes",
+			numAdds:         10,
+			numDels:         5,
+			delIndices:      []uint64{0, 2, 4, 6, 8},
+			rememberIndices: []uint64{0, 1, 2, 4, 5, 6, 8, 9},
+		},
+		{
+			name:            "delete six nodes",
+			numAdds:         10,
+			numDels:         6,
+			delIndices:      []uint64{0, 1, 3, 5, 7, 9},
+			rememberIndices: []uint64{0, 1, 2, 3, 4, 5, 6, 7, 9},
+		},
+		{
+			name:            "delete seven nodes",
+			numAdds:         10,
+			numDels:         7,
+			delIndices:      []uint64{0, 1, 2, 4, 6, 8, 9},
+			rememberIndices: []uint64{0, 1, 2, 4, 5, 6, 8, 9},
+		},
+		{
+			name:            "delete eight nodes",
+			numAdds:         10,
+			numDels:         8,
+			delIndices:      []uint64{0, 1, 2, 3, 5, 6, 8, 9},
+			rememberIndices: []uint64{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		},
+		{
+			name:            "delete nine nodes",
+			numAdds:         10,
+			numDels:         9,
+			delIndices:      []uint64{0, 1, 2, 3, 4, 6, 7, 8, 9},
+			rememberIndices: []uint64{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		},
+		{
+			name:            "delete ten nodes",
+			numAdds:         10,
+			numDels:         10,
+			delIndices:      []uint64{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+			rememberIndices: []uint64{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
No need for a `t.Run` here. You can just use the `t` that's passed in when the test starts.